arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -382,10 +382,10 @@ int arm_core_mpu_get_max_domain_partition_regions(void)
  */
 int arm_core_mpu_buffer_validate(void *addr, size_t size, int write)
 {
-	u32_t r_index;
+	s32_t r_index;
 
 	/* Iterate all mpu regions in reversed order */
-	for (r_index = _get_num_regions() + 1; r_index-- > 0;) {
+	for (r_index = _get_num_regions() - 1; r_index >= 0;  r_index--) {
```

Comment:
i'd rather not use signed.  how bout:
for (r_index = _get_num_regions() - 1; r_index; r_index--){
}

So this will iterate from the highest priority to the lowest priority.  7->0 for 8 regions, and 15->0 for 16.

Reply:
Won't this go from the closed interval `[NumRegions - 1, 0]`, to the open interval `[NumRegions - 1, 1)`?

Reply:
@agross-linaro your proposal will stop after index 1.